cli/src/cli_util.rs

```diff
@@ -3074,71 +3073,74 @@ pub fn compute_commit_location(
+                        .children()
+                        .evaluate(workspace_command.repo().as_ref())?
+                        .iter()
+                        .try_collect()?;
+
+                (after_commit_ids, new_children_ids)
+            }
+            (None, None, Some(before_commit_ids)) => {
+                let before_commits: Vec<_> = before_commit_ids
                     .iter()
-                    .commits(workspace_command.repo().store())
+                    .map(|id| workspace_command.repo().store().get_commit(id))
                     .try_collect()?;
+                // Not using `RevsetExpression::parents` here to persist the order of parents
+                // specified in `before_commits`.
+                let new_parents = before_commits
+                    .iter()
+                    .flat_map(|commit| commit.parent_ids().to_vec())
+                    .unique()
+                    .collect_vec();
 
-            (after_commits, new_children)
-        }
-        (None, None, Some(before_commits)) => {
-            // Not using `RevsetExpression::parents` here to persist the order of parents
-            // specified in `before_commits`.
-            let new_parent_ids = before_commits
-                .iter()
-                .flat_map(|commit| commit.parent_ids().iter().cloned())
-                .unique();
-            let new_parents: Vec<_> = new_parent_ids
-                .map(|commit_id| workspace_command.repo().store().get_commit(&commit_id))
-                .try_collect()?;
-
-            (new_parents, before_commits)
-        }
-        (Some(_), Some(_), _) | (Some(_), _, Some(_)) => {
-            panic!("destination cannot be used with insert_after/insert_before")
-        }
-        (None, None, None) => {
-            panic!("expected at least one of destination or insert_after/insert_before")
-        }
-    };
+                (new_parents, before_commit_ids)
+            }
+            (Some(_), Some(_), _) | (Some(_), _, Some(_)) => {
+                panic!("destination cannot be used with insert_after/insert_before")
+            }
+            (None, None, None) => {
+                panic!("expected at least one of destination or insert_after/insert_before")
+            }
+        };
 
-    if !new_children.is_empty() {
-        workspace_command.check_rewritable(new_children.iter().ids())?;
+    if !new_children_ids.is_empty() {
+        workspace_command.check_rewritable(new_children_ids.iter())?;
         ensure_no_commit_loop(
             workspace_command.repo().as_ref(),
-            &RevsetExpression::commits(new_children.iter().ids().cloned().collect_vec()),
-            &RevsetExpression::commits(new_parents.iter().ids().cloned().collect_vec()),
+            &RevsetExpression::commits(new_children_ids.iter().cloned().collect_vec()),
+            &RevsetExpression::commits(new_parent_ids.iter().cloned().collect_vec()),
```

Comment:
nit: can use `new_children_ids.clone()` and `new_parent_ids.clone()` here now, i think